src/transformers/models/jukebox/modeling_jukebox.py

```diff
@@ -0,0 +1,3133 @@
+    attn_layers = set([alignment_layer])
+    alignment_hops = {}
+    indices_hops = {}
+    # prior.to(tokens.device)
+    prior.to("cuda")
+    gc.collect()
+    torch.cuda.empty_cache()
+    for start in get_range(get_starts(total_length, n_ctx, hop_length)):
+        end = start + n_ctx
+        # set metadata offset, sample_length and lyrics tokens
+        metadata, indices_hop = prior.get_metadata(labels, start, config.sample_length, get_indices=True, offset=0)
+        metadata.to("cuda")
+        tokens_bs = torch.chunk(tokens, batch_size, dim=0)
+        metadata_bs = torch.chunk(metadata, batch_size, dim=0)
+        w_hops = []
+        for tokens_i, metadata_i in zip(tokens_bs, metadata_bs):
+            tokens_i = tokens_i.to("cuda")
+            metadata_i = metadata_i.to("cuda")
+            w_hop = prior.forward_tokens(
+                tokens_i[:, start:end], [], metadata_i, fp16=fp16, get_attn_weights=attn_layers
+            )
+            w_hops.append(w_hop[0][:, alignment_head])
+            del w_hop
+        w = torch.cat(w_hops, dim=0)
+        del w_hops
+        alignment_hop = w.float().cpu().numpy()
+        del w
+
+        # alignment_hop has shape (bs, n_ctx, nb_relevant_lyric_tokens)
+        # indices_hop is a list of len=bs, each entry of len hps.nb_relevant_lyric_tokens
+        indices_hops[start] = indices_hop
+        alignment_hops[start] = alignment_hop
+    prior.cpu()
+    gc.collect()
+    torch.cuda.empty_cache()
+
+    # Combine attn for each hop into attn for full range
+    # Use indices to place them into correct place for corresponding source tokens
+    alignments = []
+    for item in range(batch_size):
+        # Note each item has different length lyrics
+        full_tokens = labels[0, 3:]
+        alignment = np.zeros((total_length, len(full_tokens) + 1))
+        for start in reversed(get_starts(total_length, n_ctx, hop_length)):
+            end = start + n_ctx
+            alignment_hop = alignment_hops[start][item]
+            indices = indices_hops[start][item]
+            alignment[start:end, indices] = alignment_hop
+        alignment = alignment[: total_length - padding_length, :-1]  # remove token padding, and last lyric index
+        alignments.append(alignment)
+    return alignments
+
+
+def save_wav(fname, lvl, metas, aud, sampling_rate):
+    import soundfile
+
+    aud = torch.clamp(aud, -1, 1).cpu().numpy()
+    for i in list(range(aud.shape[0])):
+        if metas is not None:
+            # twitter prompts or inputs are in the form of a dictionnary
+            artists, genres, lyrics = list(metas)[i].values()
+            path = f"{fname}/lvl_{lvl}-{artists}-{genres}-{lyrics[:5]}-{i}.wav"
+            soundfile.write(path, aud[i], samplerate=sampling_rate, format="wav")
+        else:
+            soundfile.write(f"{fname}/lvl_{lvl}-sample-{i}.wav", aud[i], samplerate=sampling_rate, format="wav")
+
+
+def load_audio(file, sampling_rate, offset, duration, mono=False):
```

Comment:
this should also definitely not be in `transformers` I don't want to get librosa issues in Transformers